cargo-shuttle/tests/integration/run.rs

```diff
@@ -359,3 +360,68 @@ async fn thruster_hello_world() {
 
     assert_eq!(request_text, "Hello, World!");
 }
+
+/// creates a `cargo-shuttle` run instance with some reasonable defaults set, but using the router IP instead.
+async fn cargo_shuttle_run_with_router_ip(working_directory: &str) -> u16 {
+    let working_directory = canonicalize(working_directory).unwrap();
+    let port = pick_unused_port().unwrap();
+    let ip = local_ip().unwrap();
+    let run_args = RunArgs { port, router_ip: true };
```

Comment:
Can you find a way to remove the duplicate code between this new function and `cargo_shuttle_run`?

Maybe a new function like:
```rust
async fn new_function(working_directory: &str, router_ip: bool) -> u16 {
...
}
```

That gets called by both `cargo_shuttle_run` and `cargo_shuttle_run_with_router_ip`.